File: video-service-client/src/main/java/com/boclips/videos/service/client/VideoType.java
package com.boclips.videos.service.client;

import java.util.Arrays;

public enum VideoType {
    NEWS(1),
    STOCK(2),
    INSTRUCTIONAL_CLIPS(3);

    private final Integer id;

    VideoType(Integer id) {
        this.id = id;
    }

    public static VideoType fromId(Integer id) {
        return Arrays.stream(VideoType.values()).filter(videoType -> videoType.id.equals(id))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Video type id not recognised: " + id));
    }
}

File: video-service/src/main/kotlin/com/boclips/videos/service/domain/model/asset/VideoType.kt
package com.boclips.videos.service.domain.model.asset

enum class VideoType(val id: Int) {
    OTHER(0),
    NEWS(1),
    STOCK(2),
    INSTRUCTIONAL_CLIPS(3),
    TV_CLIPS(4),
    NEWS_PACKAGE(5),
    UGC_NEWS(6),
    VR_360_STOCK(7),
    VR_360_IMMERSIVE(8),
    SHORT_PROGRAMME(9),
    TED_TALKS(10),
    TED_ED(11);

    companion object {

        private val typeById = VideoType.values().map { it.id to it }.toMap()

        fun fromId(id: Int): VideoType {
            return typeById[id] ?: throw IllegalArgumentException("The type id $id is invalid")
        }
    }
}